starknet_py/tests/e2e/mock/compile_contracts.sh

```diff
@@ -1,32 +1,70 @@
 #!/bin/bash
 
-MOCK_DIRECTORY=starknet_py/tests/e2e/mock
-CONTRACTS_DIRECTORY="$MOCK_DIRECTORY"/contracts
-CONTRACTS_COMPILED_DIRECTORY="$MOCK_DIRECTORY"/contracts_compiled
+compile_contracts_with_scarb() {
+    if ! command -v asdf >/dev/null 2>&1; then
+        echo "Install asdf, before executing this script is required to manage scarb version"
+        exit 1
+    fi
 
-# delete all artifacts except precompiled ones
-find $CONTRACTS_COMPILED_DIRECTORY -maxdepth 1 -type f -delete
+    asdf plugin-add scarb
+    asdf install scarb 0.4.1
+    asdf install scarb 2.6.3
```

Comment:
it may be troublesome to remember all the places that define the versions. Ideally, this script would hold paths to .tool_versions and take them directly from the file, so you only change scarb version in one place